src/components/Resource/Forms/TrainingSelectFile.tsx

```diff
@@ -0,0 +1,311 @@
+            Select Epoch #{epoch.epoch_number}
+          </Button>
+          <DownloadButton component="a" canDownload href={epoch.model_url}>
+            <Text align="center">
+              {/*{`Download (${formatKBytes(modalData.file?.sizeKB)})`}*/}
+              Download
+            </Text>
+          </DownloadButton>
+        </Group>
+        <Group className={classes.epochRow} style={{ justifyContent: 'space-evenly' }}>
+          {epoch.sample_images && epoch.sample_images.length > 0 ? (
+            epoch.sample_images.map((imgData, index) => (
+              <Stack key={index} style={{ justifyContent: 'flex-start' }}>
+                {/*<div className={classes.imgOverlay}>*/}
+                <Image
+                  alt={`Sample image #${index}`}
+                  src={imgData.image_url}
+                  imageProps={{
+                    style: {
+                      height: '200px',
+                      // if we want to show full image, change objectFit to contain
+                      objectFit: 'cover',
+                      // object-position: top;
+                      width: '100%',
+                    },
+                  }}
+                />
+                {/*</div>*/}
+                <Textarea autosize minRows={1} maxRows={4} value={imgData.prompt} readOnly />
+              </Stack>
+            ))
+          ) : (
+            <Center>
+              <Text>No images available.</Text>
+            </Center>
+          )}
+        </Group>
+      </Stack>
+    </Paper>
+  );
+};
+
+export default function TrainingSelectFile({
+  model,
+  onBackClick,
+  onNextClick,
+}: {
+  model?: ModelWithTags;
+  onBackClick: () => void;
+  onNextClick: () => void;
+}) {
+  const queryUtils = trpc.useContext();
+  const { upload } = useS3UploadStore();
+  const [awaitInvalidate, setAwaitInvalidate] = useState<boolean>(false);
+
+  console.log(model);
+  const modelVersion = model?.modelVersions?.[0];
+  console.log(modelVersion);
+  // TODO [bw] need to worry about multiple files? which one will this grab?
+  const modelFile = modelVersion?.files.find((f) => f.type === 'Training Data');
+  console.log(modelFile);
+
+  // TODO [bw] autopopulate based on already selected model. how best to do this?
+  const [selectedFile, setSelectedFile] = useState<string | undefined>();
+
+  const notificationId = `${modelVersion?.id || 1}-uploading-file-notification`;
+
+  const createFileMutation = trpc.modelFile.create.useMutation({
+    async onSuccess() {
+      if (!model || !modelVersion) {
+        showErrorNotification({
+          error: new Error('Missing model data. Please try again.'),
+        });
+        return;
+      }
+
+      const versionMutateData = {
+        id: modelVersion.id,
+        modelId: model.id,
+        name: model.name,
+        baseModel: 'SD 1.5' as const,
+        trainingStatus: TrainingStatus.Approved,
+      };
+
+      upsertVersionMutation.mutate(versionMutateData);
```

Comment:
Could it be possible to do this at server level in the same endpoint that we use to create the model file? This way we avoid having to make another request

Reply:
i would have to modify that entire function to handle this case...can't think of a fast way to do that honestly.